### PR TITLE
Feature/222284 lingo re branding and UI update

### PR DESCRIPTION
**Pull request by samridhis**
latest UI changes

### CODE REVIEW

File: res/css/views/auth/_AuthHeader.scss [hunk 1]
```diff
@@ -20,6 +20,7 @@ limitations under the License.
     width: 206px;
     padding: 25px 25px;
     box-sizing: border-box;
+    background-color: white!important;
```
Comment (jrschremmer) [inline res/css/views/auth/_AuthHeader.scss]: Hi Samridhi,  please review how we have been implementing branding (e.g. https://dev.azure.com/ACTGovernment/Pegacorn/_workitems/edit/219811).  As we want to be able to push our fork back to element at some stage (to minimise merge issues when we pull from element master when required) we can't change the default behaviour of the code.  Instead either implement the change via:
1. The "Pegacorn communicate - Light theme custom colors" custom theme
2. Controlled by a config.json configuration property that defaults to the current value in the element code, if the propoerty is not in config.json
3. We have our own copy of the file in aether-host-files/common/element-web/matrix-react-sdk (this is the least favoured solution as it doesn't address the merge issues with element master code, but has been done for a quick tactical solution)

**Review by jrschremmer — CHANGES_REQUESTED** [review]

File: src/components/views/auth/PasswordLogin.tsx [hunk 2]
```diff
@@ -484,7 +484,7 @@ export default class PasswordLogin extends React.PureComponent<IProps, IState> {
                     {forgotPasswordJsx}
                     { !this.props.busy && <input className="mx_Login_submit"
                         type="submit"
-                        value={_t('Sign in')}
+                        value={('Log in')}
```
Comment (jrschremmer) [inline src/components/views/auth/PasswordLogin.tsx]: This breaks internationalisation in element.  I'd suggest updating the value in src\i18n\strings\en_EN.json

File: src/components/views/elements/SSOButtons.tsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ const SSOButton: React.FC<ISSOButtonProps> = ({
     mini,
     ...props
 }) => {
-    const label = idp ? _t("Continue with %(provider)s", { provider: idp.name }) : _t("Sign in with single sign-on");
+    const label = idp ? _t("Continue with %(provider)s", { provider: idp.name }) : ("Log in with single sign-on");
```
Comment (jrschremmer) [inline src/components/views/elements/SSOButtons.tsx]: This breaks internationalisation in element. I'd suggest updating the value in src\i18n\strings\en_EN.json

File: res/css/views/auth/_AuthHeader.scss (hunk 1 above)
Comment (samridhis) [reply res/css/views/auth/_AuthHeader.scss]: image
I just went through changes from Surendra on the changeset. I am able to see his changes on the same file.

Comment (jrschremmer) [reply res/css/views/auth/_AuthHeader.scss]: Yes but that file is in a different git repository, which is option 3 I described above.  If you implement your changes in that repo and revert the changes in this repo, that will be fine.

File: src/components/structures/HomePage.tsx [hunk 4]
```diff
@@ -119,10 +119,7 @@ const HomePage: React.FC<IProps> = ({ justRegistered = false }) => {
        }
 
         introSection = <React.Fragment>
-            <img src={logoUrl} alt={config.brand} />
-            {showLiberateYourCommunicationText &&
```
Comment (jrschremmer) [inline src/components/structures/HomePage.tsx:127]: Why has the "showLiberateYourCommunicationText &&" block of code been removed?  If we don't want to show this text, it can be controlled by configuration as implemented correctly by Surendra.

File: src/components/structures/HomePage.tsx [hunk 5]
```diff
@@ -119,14 +119,7 @@ const HomePage: React.FC<IProps> = ({ justRegistered = false }) => {
        }
 
         introSection = <React.Fragment>
-            {customConfig.showSecondaryLogoInAuthenticatedScreen && <img src={logoUrlSecondary} style={logoSecondaryStyle} alt={logoSecondaryDescription} />}
-            {customConfig.showPrimaryLogoInAuthenticatedScreen &&
-            <img src={logoUrl} alt={config.brand}/>
-            }
-            {showWelcomeToElementText ?
```
Comment (jrschremmer) [inline src/components/structures/HomePage.tsx]: Why has the "showWelcomeToElementText" block of code been removed? If we don't want to show this text, it can be controlled by configuration as implemented correctly by Surendra.

File: src/components/structures/TabbedView.tsx [hunk 6]
```diff
@@ -125,6 +125,14 @@ export default class TabbedView extends React.Component<IProps, IState> {
                     {labels}
                 </div>
                 {panel}
+                <div className="mx_AuthFooter_TabbedView">
```
Comment (jrschremmer) [inline src/components/structures/TabbedView.tsx]: Please conditionally show this block of code based on a configuration property (my suggestion would be  AuthFooterTabbedViewImgPath) having a value, that defaults to null, so the standard element behaviour isn't changed.  We can set the configuration property to themes/element/img/logos/ACTGov_inline_rev_black.png to show it for us.  The alt text needs to also be externalised into the configuration file.

File: src/components/views/auth/AuthHeader.js [hunk 7]
```diff
@@ -29,7 +29,12 @@ export default class AuthHeader extends React.Component {
         const LanguageSelector = sdk.getComponent('views.auth.LanguageSelector');
 
         return (
-            <div className="mx_AuthHeader">
+            <div className="mx_AuthHeader"> 
+                <br/>
```
Comment (jrschremmer) [inline src/components/views/auth/AuthHeader.js]: Please conditionally show this block of code based on a configuration property (my suggestion would be AuthHeaderImgPath) having a value, that defaults to null, so the standard element behaviour isn't changed. We can set the configuration property to themes/element/img/logos/ACT-Government-inline-back-trans.jpg to show it for us.  The alt text needs to also be externalised into the configuration file.

**Review by jrschremmer — CHANGES_REQUESTED** [review]

File: res/css/views/auth/_AuthHeader.scss (hunk 1 above)
Comment (samridhis) [reply res/css/views/auth/_AuthHeader.scss]: Thanks, I have reverted this specific change as it was already a part of aether host files.

File: src/components/structures/TabbedView.tsx (hunk 6 above)
Comment (samridhis) [reply src/components/structures/TabbedView.tsx]: Done

File: src/components/structures/HomePage.tsx (hunk 5 above)
Comment (samridhis) [reply src/components/structures/HomePage.tsx]: Done

File: src/components/structures/HomePage.tsx (hunk 4 above)
Comment (samridhis) [reply src/components/structures/HomePage.tsx:127]: Done

File: src/components/views/auth/AuthHeader.js (hunk 7 above)
Comment (samridhis) [reply src/components/views/auth/AuthHeader.js]: Done

File: src/components/views/auth/AuthHeader.js [hunk 8]
```diff
@@ -27,13 +28,16 @@ export default class AuthHeader extends React.Component {
     render() {
         const AuthHeaderLogo = sdk.getComponent('auth.AuthHeaderLogo');
         const LanguageSelector = sdk.getComponent('views.auth.LanguageSelector');
+        const loginScreen = SdkConfig.get().loginScreen;
+        const showAuthHeaderLogo_Act = loginScreen?.displayActLogo ?? true;
```
Comment (jrschremmer) [inline src/components/views/auth/AuthHeader.js]: Please remove Act from the config property name and the variable name.  I think showAuthHeaderLogo should default to false e.g. ?? false, so the default behaviour of element is not affected.  Also please be consistent with the current code and centralise this into src/config.ts

File: src/components/structures/TabbedView.tsx [hunk 9]
```diff
@@ -118,13 +119,24 @@ export default class TabbedView extends React.Component<IProps, IState> {
...
         const panel = this._renderTabPanel(this.props.tabs[this._getActiveTabIndex()]);
-
+        
         return (
             <div className="mx_TabbedView">
                 <div className="mx_TabbedView_tabLabels">
                     {labels}
                 </div>
                 {panel}
+                {customConfig.tabbedViewDisplayActLogo === true &&
```
Comment (jrschremmer) [inline src/components/structures/TabbedView.tsx]: You haven't externalised themes/element/img/logos/ACTGov_inline_rev_black.png and the alt text into the configuration file, as discussed in the previous conversation (I was unable to reply in that conversation in the github UI).  Also please remove Act from the config name.  Also please add the description of any new config values to feature-communicate-list.md

File: res/css/structures/_HomePage.scss [hunk 10]
```diff
@@ -33,7 +33,7 @@ limitations under the License.
     }
 
     img {
-        height: 48px;
+        height: 120px;
```
Comment (jrschremmer) [inline res/css/structures/_HomePage.scss]: As mentioned in another conversation, please implement all changes as either:

1. Controlled by a config.json configuration property that defaults to the current value in the element code, if the propoerty is not in config.json
2. We have our own copy of the file in aether-host-files/common/element-web/matrix-react-sdk (this is the least favoured solution as it doesn't address the merge issues with element master code, but has been done for a quick tactical solution)

File: res/css/views/auth/_AuthFooter.scss [hunk 11]
```diff
@@ -15,7 +15,7 @@ limitations under the License.
 */
 
 .mx_AuthFooter {
-    text-align: center;
+    text-align: right;
```
Comment (jrschremmer) [inline res/css/views/auth/_AuthFooter.scss]: As mentioned in another conversation, please implement all changes as either:
1. Controlled by a config.json configuration property that defaults to the current value in the element code, if the propoerty is not in config.json
2. We have our own copy of the file in aether-host-files/common/element-web/matrix-react-sdk (this is the least favoured solution as it doesn't address the merge issues with element master code, but has been done for a quick tactical solution)

File: res/css/views/auth/_AuthHeaderLogo.scss [hunk 12]
```diff
@@ -21,11 +21,30 @@ limitations under the License.
 }
 
 .mx_AuthHeaderLogo img {
-    width: 100%;
+    //width: 100%;
```
Comment (jrschremmer) [inline res/css/views/auth/_AuthHeaderLogo.scss]: As mentioned in another conversation, please implement all changes as either:
1. Controlled by a config.json configuration property that defaults to the current value in the element code, if the propoerty is not in config.json
2. We have our own copy of the file in aether-host-files/common/element-web/matrix-react-sdk (this is the least favoured solution as it doesn't address the merge issues with element master code, but has been done for a quick tactical solution)

File: res/css/views/auth/_AuthPage.scss [hunk 13]
```diff
@@ -19,7 +19,7 @@ limitations under the License.
     min-height: 100%;
     display: flex;
     flex-direction: column;
-    background-color: $authpage-bg-color;
+    background-color: #002677!important;
```
Comment (jrschremmer) [inline res/css/views/auth/_AuthPage.scss]: I suggest you change the value of $authpage-bg-color in our copy of themes\light\css\_light.scss in aether-host-files and avoid !important, as it should only be used as a last resort.

File: res/css/structures/_TabbedView.scss [hunk 14]
```diff
@@ -27,7 +27,9 @@ limitations under the License.
     right: 0;
     margin-top: 8px;
 }
-
+.mx_AuthFooter_Image_Tabbed{
+height: 120px!important;
```
Comment (jrschremmer) [inline res/css/structures/_TabbedView.scss]: Is !important required here ... it should only be used as a last resort.  NOTE: this change doesn't have to move to aether-host-files as it is adding new code and not making any updates.

File: res/themes/light/css/_light.scss [hunk 15]
```diff
@@ -14,7 +14,7 @@ $monospace-font-family: Inconsolata, Twemoji, 'Apple Color Emoji', 'Segoe UI Emo
 
 // unified palette
 // try to use these colors when possible
-$accent-color: #0DBD8B;
+$accent-color:  #002677;
```
Comment (jrschremmer) [inline res/themes/light/css/_light.scss]: Please move this change to aether-host-files

File: src/components/views/auth/AuthHeader.js [hunk 16]
```diff
@@ -27,9 +28,16 @@ export default class AuthHeader extends React.Component {
...
 
         return (
-            <div className="mx_AuthHeader">
+            <div className="mx_AuthHeader"> 
+                {showAuthHeaderLogo_Act &&                  
+                <div>
+                    <br/>              
+                    <img className="mx_AuthHeaderLogo_Act" src="themes/element/img/logos/ACT-Government-inline-back-trans.jpg" alt="ACT Govt Logo"  />
```
Comment (jrschremmer) [inline src/components/views/auth/AuthHeader.js]: As mentioned in another comment, please externalised the img src path and the alt text into the configuration file and remove any references to "Act"

**Review by jrschremmer — CHANGES_REQUESTED** [review]

File: src/components/views/auth/AuthHeader.js (hunk 16 above)
Comment (samridhis) [reply src/components/views/auth/AuthHeader.js]: Done

File: res/css/structures/_TabbedView.scss (hunk 14 above)
Comment (samridhis) [reply res/css/structures/_TabbedView.scss]: Done

File: res/css/structures/_HomePage.scss (hunk 10 above)
Comment (samridhis) [reply res/css/structures/_HomePage.scss]: done

File: res/css/views/auth/_AuthFooter.scss (hunk 11 above)
Comment (samridhis) [reply res/css/views/auth/_AuthFooter.scss]: done

File: res/css/views/auth/_AuthHeaderLogo.scss (hunk 12 above)
Comment (samridhis) [reply res/css/views/auth/_AuthHeaderLogo.scss]: Current style is set as below in this style width is repeacted twice. I am not sure what is the benifit of doing that ? 
.mx_AuthHeaderLogo img {
    width: 100%;
    width: 150px;
    height: 100px;
}

File: res/css/views/auth/_AuthFooter.scss (hunk 11 above)
Comment (jrschremmer) [reply res/css/views/auth/_AuthFooter.scss]: Hi Sam, this change is still hard-coded in the code.  Please externalise it into the theme like $font-14px is.

File: res/css/views/auth/_AuthHeaderLogo.scss [hunk 17]
```diff
@@ -22,10 +22,29 @@ limitations under the License.
 
 .mx_AuthHeaderLogo img {
     width: 100%;
+    width: 150px;
```
Comment (jrschremmer) [inline res/css/views/auth/_AuthHeaderLogo.scss]: There are 2 definitions for width here, please externalise a single value into the themes.  The values in the element themes will be 100% and our copy of the light theme in aether-host-files can specify 150px.  For the height value the value in the element themes, you will have to work out as based on https://developer.mozilla.org/en-US/docs/Web/CSS/height I'm not sure if previously having no value is equivalent to inherit, unset or auto.  In our copy of the light theme the value would be 100px

File: res/css/views/auth/_AuthPage.scss (hunk 13 above)
Comment (jrschremmer) [reply res/css/views/auth/_AuthPage.scss]: This comment is still unresolved

File: res/css/views/elements/_SSOButtons.scss [hunk 18]
```diff
@@ -46,13 +46,13 @@ limitations under the License.
     }
 
     .mx_SSOButton_default {
-        color: $button-primary-bg-color;
-        background-color: $button-secondary-bg-color;
-        border-color: $button-primary-bg-color;
+        color: $button-secondary-bg-color;
```
Comment (jrschremmer) [inline res/css/views/elements/_SSOButtons.scss:59]: These changes look odd, as the default button is being assigned the secondary colours which is not the expected behaviour.

File: res/css/views/elements/_SSOButtons.scss [hunk 19]
```diff
@@ -46,13 +46,13 @@ limitations under the License.
...
+        background-color: $button-primary-bg-color;
+        border-color: $button-secondary-bg-color;
     }
     .mx_SSOButton_default.mx_SSOButton_primary {
-        color: $button-primary-fg-color;
-        background-color: $button-primary-bg-color;
+        color: $button-primary-bg-color;
```
Comment (jrschremmer) [inline res/css/views/elements/_SSOButtons.scss:64]: This change looks odd as the foreground and background colours have been swapped?

File: res/themes/light/css/_light.scss [hunk 20]
```diff
@@ -14,7 +14,7 @@ $monospace-font-family: Inconsolata, Twemoji, 'Apple Color Emoji', 'Segoe UI Emo
 
 // unified palette
 // try to use these colors when possible
-$accent-color: #0DBD8B;
+$accent-color:  #00818b;
```
Comment (jrschremmer) [inline res/themes/light/css/_light.scss]: Changes to this file need to be reverted, so we don't change the default behaviour of element.  Instead these changes go into our copy of this file in aether-host-files

File: src/components/structures/TabbedView.tsx [hunk 21]
```diff
@@ -118,13 +119,24 @@ export default class TabbedView extends React.Component<IProps, IState> {
...
                 <div className="mx_TabbedView_tabLabels">
                     {labels}
                 </div>
                 {panel}
+                {customConfig.tabbedViewdisplaySecondaryLogo === true &&
+                    <div className="mx_AuthFooter_TabbedView">
+                        <a href="" target="_blank" rel="noreferrer noopener">                        
+                        </a>
+                        <span className="mx_AuthFooter_brand_TabbedView">
+                            <img className="mx_AuthFooter_Image_Tabbed" src="themes/element/img/logos/ACTGov_inline_rev_black.png" alt="ACT Govt Logo"/>
```
Comment (jrschremmer) [inline src/components/structures/TabbedView.tsx]: The image path and ACT Govt Logo need to be externalised into the config too pls

File: src/components/views/auth/PasswordLogin.tsx [hunk 22]
```diff
@@ -433,7 +433,8 @@ export default class PasswordLogin extends React.PureComponent<IProps, IState> {
         const loginField = this.renderLoginField(this.state.loginType, !autoFocusPassword);
 
         let loginType;
-        if (!SdkConfig.get().disable_3pid_login) {
+        const disable_3pid_login =SdkConfig.get().loginScreen?.disable_3pid_login??false;
```
Comment (jrschremmer) [inline src/components/views/auth/PasswordLogin.tsx]: Please revert this change and continue to use the element default config location for disable_3pid_login

File: src/components/views/auth/RegistrationForm.tsx [hunk 23]
```diff
@@ -411,7 +411,7 @@ export default class RegistrationForm extends React.PureComponent<IProps, IState
     }
 
     private showPhoneNumber() {
-        const threePidLogin = !SdkConfig.get().disable_3pid_login;
+        const threePidLogin = !SdkConfig.get().loginScreen?.disable_3pid_login;
```
Comment (jrschremmer) [inline src/components/views/auth/RegistrationForm.tsx]: Please revert this change and continue to use the element default config location for disable_3pid_login

File: src/config.ts [hunk 24]
```diff
@@ -20,7 +25,7 @@ export const signInTextNeedsToBeReplaced = loginScreen?.changeSigninToLoginTextL
 
 // background color on login screen
 // used by app-web to load a colored background instead of lake image
-export const loginScreenBackgroundColor = loginScreen?.backgroundColor ?? null;
+export const showLoginScreenBackgroundImage = loginScreen?.showLoginScreenBackgroundImage ?? null;
```
Comment (jrschremmer) [inline src/config.ts]: Searching this pull request, I can't see where showLoginScreenBackgroundImage is still used?

File: src/config.ts [hunk 25]
```diff
@@ -43,3 +48,34 @@ export const showHomeServerDetail = loginScreen?.showHomeServerInfo ?? true;
...
+// show/hide liberate your communication text
+export const showLiberateYourCommunicationText = authenticatedHomeScreen?.showLiberateYourCommunicationText ?? true;
+
+//show/hide welcome to {brand} text
+export const showWelcomeToElementText = authenticatedHomeScreen?.showWelcomeToElementText ?? true;
+
+// show primary logo on logged in screen
+// Due to positioning and etc this switch would make it easy to have that logo configured.
+// it is to control logo that element provides
+export const showPrimaryLogoOnLoginScreen = logoConfig?.showPrimaryLogoOnLoginScreen ?? true;
```
Comment (jrschremmer) [inline src/config.ts]: I can't see where the showPrimaryLogo* config items are still used?  Please review all the config items in this file and any still used please ensure their description is added to feature-communicate-list.md

**Review by jrschremmer — CHANGES_REQUESTED** [review]

File: src/config.ts (hunk 24 above)
Comment (samridhis) [reply src/config.ts]: done

File: src/config.ts (hunk 25 above)
Comment (samridhis) [reply src/config.ts]: done

File: src/components/views/auth/RegistrationForm.tsx (hunk 23 above)
Comment (samridhis) [reply src/components/views/auth/RegistrationForm.tsx]: @jrschremmer I am not sure why to revert this change. there are multiple configurations inside the login screen which i have kept into the login screen config. it will be easier for a person to identify the configuration by that.
 "loginScreen": {
        "displaySecondaryLogo":false,
        "displaySecondaryVerticalLogo":false,        
        "displayLanguageSelector":false,    
        "disable_3pid_login": false,            
        "showHomeServerInfo": false,
        "showMatrixDefaultFooterLinks": false,
        "allowPhoneBasedAuthentication": false,
        "allowEmailBasedAuthentication": false,
        "changeSigninToLoginTextLabel": true,
        "backgroundColor": "(0,38, 119)",
        "logo":{
                "img":{
                "width": "150px",
                "height": "100px"
            }
        }

File: src/components/views/auth/PasswordLogin.tsx (hunk 22 above)
Comment (samridhis) [reply src/components/views/auth/PasswordLogin.tsx]: I am not sure why to revert this change. there are multiple configurations inside the login screen which i have kept into the login screen config. it will be easier for a person to identify the configuration by that.
"loginScreen": {
"displaySecondaryLogo":false,
"displaySecondaryVerticalLogo":false,
"displayLanguageSelector":false,
"disable_3pid_login": false,
"showHomeServerInfo": false,
"showMatrixDefaultFooterLinks": false,
"allowPhoneBasedAuthentication": false,
"allowEmailBasedAuthentication": false,
"changeSigninToLoginTextLabel": true,
"backgroundColor": "(0,38, 119)",
"logo":{
"img":{
"width": "150px",
"height": "100px"
}
}

File: src/components/structures/TabbedView.tsx (hunk 9 above)
Comment (samridhis) [reply src/components/structures/TabbedView.tsx]: done

File: src/components/views/auth/AuthHeader.js (hunk 8 above)
Comment (samridhis) [reply src/components/views/auth/AuthHeader.js]: done

File: res/css/views/auth/_AuthPage.scss (hunk 13 above)
Comment (samridhis) [reply res/css/views/auth/_AuthPage.scss]: done

File: res/css/views/auth/_AuthFooter.scss (hunk 11 above)
Comment (samridhis) [reply res/css/views/auth/_AuthFooter.scss]: sorry done now

File: res/css/views/auth/_AuthHeaderLogo.scss (hunk 12 above)
Comment (samridhis) [reply res/css/views/auth/_AuthHeaderLogo.scss]: done

File: res/themes/light/css/_light.scss (hunk 15 above)
Comment (samridhis) [reply res/themes/light/css/_light.scss]: done

File: res/css/views/elements/_SSOButtons.scss (hunk 19 above)
Comment (samridhis) [reply res/css/views/elements/_SSOButtons.scss:64]: it's all good

Comment (samridhis) [reply res/css/views/elements/_SSOButtons.scss:64]: this change needed for light color button now instead of white button previously for SSO

File: res/css/views/elements/_SSOButtons.scss (hunk 18 above)
Comment (samridhis) [reply res/css/views/elements/_SSOButtons.scss:59]: this change needed for the light color button now instead of the white button previously for SSO

File: res/css/views/auth/_AuthHeaderLogo.scss (hunk 17 above)
Comment (samridhis) [reply res/css/views/auth/_AuthHeaderLogo.scss]: to avoid confusion i have added this file in aether host files with the changes..

File: res/css/views/elements/_SSOButtons.scss (hunk 18 above)
Comment (jrschremmer) [reply res/css/views/elements/_SSOButtons.scss:59]: But you have changed the default element behaviour, which we need to avoid ... changing the value in our custom theme of $button-primary-bg-color to the value of $button-secondary-bg-color maybe a solution but that could have other unintended consequences ... perhaps based on config change the style name used e.g. instead of using .mx_SSOButton_default use .mx_SSOButton_default_inverse which you will need to add.

File: res/css/views/elements/_SSOButtons.scss (hunk 19 above)
Comment (jrschremmer) [reply res/css/views/elements/_SSOButtons.scss:64]: Same as the other comment on the file.

File: res/themes/light/css/_light.scss (hunk 20 above)
Comment (samridhis) [reply res/themes/light/css/_light.scss]: done

File: src/components/views/auth/PasswordLogin.tsx (hunk 22 above)
Comment (jrschremmer) [reply src/components/views/auth/PasswordLogin.tsx]: Because you are changing the default element behaviour, which we are trying to avoid

File: src/config.ts [hunk 26]
```diff
@@ -43,3 +46,19 @@ export const showHomeServerDetail = loginScreen?.showHomeServerInfo ?? true;
...
+export const showWelcomeToElementText = authenticatedHomeScreen?.showWelcomeToElementText ?? true;
+
+//find logo secondary
+export const logoSecondary = logoConfig?.logo_secondary ?? null;
+
+
+// tab condition
+export const tabbedView =config['tabbedView'];
+export const tabbedViewdisplaySecondaryLogo  = tabbedView?.displaySecondaryLogo ?? false;
+export const logoUrlTabbedViewFooter =  tabbedView?.imgUrl;
```
Comment (jrschremmer) [inline src/config.ts]: For consistency please rename 
imgUrl to secondaryLogoUrl
description to secondaryLogoAltText

**Review by jrschremmer — CHANGES_REQUESTED** [review]

File: src/components/views/auth/PasswordLogin.tsx (hunk 22 above)
Comment (samridhis) [reply src/components/views/auth/PasswordLogin.tsx]: done

File: src/components/views/auth/RegistrationForm.tsx (hunk 23 above)
Comment (samridhis) [reply src/components/views/auth/RegistrationForm.tsx]: done

File: src/components/structures/TabbedView.tsx [hunk 27]
```diff
@@ -118,13 +119,23 @@ export default class TabbedView extends React.Component<IProps, IState> {
...
+        const logoUrlAltText =  customConfig.logoDescTabbedViewFooter;
         return (
             <div className="mx_TabbedView">
                 <div className="mx_TabbedView_tabLabels">
                     {labels}
                 </div>
                 {panel}
+                {customConfig.tabbedViewdisplaySecondaryLogo === true &&
+                    <div className="mx_AuthFooter_TabbedView">
+                        <a href="" target="_blank" rel="noreferrer noopener">
```
Comment (jrschremmer) [inline src/components/structures/TabbedView.tsx]: What is this empty hyperlink for?  If it is no longer required pls remove it

Comment (samridhis) [reply src/components/structures/TabbedView.tsx]: Sure

Comment (samridhis) [reply src/components/structures/TabbedView.tsx]: done

File: src/config.ts (hunk 26 above)
Comment (samridhis) [reply src/config.ts]: done

File: res/css/views/elements/_SSOButtons.scss (hunk 18 above)
Comment (samridhis) [reply res/css/views/elements/_SSOButtons.scss:59]: done

File: res/css/views/elements/_SSOButtons.scss (hunk 19 above)
Comment (samridhis) [reply res/css/views/elements/_SSOButtons.scss:64]: done

**Review by jrschremmer — APPROVED** [review]
